north/src/runtime/linux/device_mapper.rs

```diff
@@ -192,11 +199,16 @@ impl Dm {
     /// used.
     ///
     /// Valid flags: DM_DEFERRED_REMOVE
-    pub async fn device_remove(&self, id: &str, options: &DmOptions) -> Result<DeviceInfo> {
+    pub async fn device_remove(
+        &self,
+        id: &str,
+        options: &DmOptions,
+    ) -> Result<DeviceInfo, DeviceMapperError> {
         let mut hdr = options.to_ioctl_hdr(Some(id), DmFlags::DM_DEFERRED_REMOVE);
 
         self.do_ioctl(DM_DEV_REMOVE_CMD as u8, &mut hdr, None)
-            .await?;
+            .await
+            .map_err(|_| DeviceMapperError::DeviceRemovalFailed);
```

Comment:
This elides the original io::Error. `DeviceMapperError` should wrap a `io::Error`